test/Removal.test.ts

```diff
@@ -0,0 +1,162 @@
+import type { BigNumber } from 'ethers';
+
+import type { Contracts } from '@/test/helpers';
+import { getDeployments, expect, hardhat } from '@/test/helpers';
+import { hre } from '@/utils/hre';
+
+const setupTest = hre.deployments.createFixture(
+  async (): Promise<Contracts> => {
+    await hre.deployments.fixture(); // ensure you start from a fresh deployments
+    return getDeployments({ hre });
+  }
+);
+
+describe('Removal', () => {
+  describe('Minting removals', () => {
+    it('should mint a batch of removals without listing any', async () => {
+      const { fifoMarket, removal } = await setupTest();
+      const { supplier } = await hardhat.getNamedAccounts();
```

Comment:
You can use

```ts
const supplier = hre.namedAccounts.supplier;
```

With the newly extended environment (see extendEnvironment) which awaits this promise and the signers promise once so we don't have to do this every time

Reply:
nice! replaced this everywhere relevant